src/pages/stacks-blockchain/integrate-stacking.md

```diff
@@ -0,0 +1,414 @@
+  broadcastTransaction,
+  standardPrincipalCV,
+} = require('@blockstack/stacks-transactions');
+const {
+  InfoApi,
+  AccountsApi,
+  SmartContractsApi,
+  Configuration,
+  TransactionsApi,
+} = require('@stacks/blockchain-api-client');
+const c32 = require('c32check');
+
+const apiConfig = new Configuration({
+  fetchApi: fetch,
+  basePath: 'https://stacks-node-api.blockstack.org',
+});
+
+// generate rnadom key
+const privateKey = makeRandomPrivKey();
+
+// get Stacks address
+const stxAddress = getAddressFromPrivateKey(
+  privateKeyToString(privateKey),
+  TransactionVersion.Testnet
+);
+```
+
+-> Review the [accounts guide](/stacks-blockchain/accounts) for more details
+
+## Step 3: Display stacking info
+
+In order to inform users about the upcoming reward cycle, we need to obtain Stacking information:
+
+```js
+const info = new InfoApi(apiConfig);
+
+const poxInfo = await info.getPoxInfo();
+const coreInfo = await info.getCoreApiInfo();
+const blocktimeInfo = await info.getNetworkBlockTimes();
+
+console.log({ poxInfo, coreInfo, blocktimeInfo });
+```
+
+-> Check out the API references for the 3 endpoints used here: [GET /v2/info](https://blockstack.github.io/stacks-blockchain-api/#operation/get_core_api_info), [GET v2/pox](https://blockstack.github.io/stacks-blockchain-api/#operation/get_pox_info), and [GET /extended/v1/info/network_block_times](https://blockstack.github.io/stacks-blockchain-api/#operation/get_network_block_times)
+
+The object, including PoX, core, and block time information, looks like this:
+
+```js
+{
+  poxInfo: {
+    contract_id: 'ST000000000000000000002AMW42H.pox',
+    first_burnchain_block_height: 0,
+    min_amount_ustx: 2000000000000,
+    registration_window_length: undefined,
+    rejection_fraction: 25,
+    reward_cycle_id: 4,
+    reward_cycle_length: 120
+  },
+  coreInfo: {
+    limit: undefined,
+    peer_version: 385875968,
+    burn_consensus: undefined,
+    burn_block_height: 605,
+    stable_burn_consensus: undefined,
+    stable_burn_block_height: 604,
+    server_version: 'blockstack-core 0.0.1 => 23.0.0.0 (master:5b816c2+, release build, linux [x86_64])',
+    network_id: 2147483648,
+    parent_network_id: 3669344250,
+    stacks_tip_height: 104,
+    stacks_tip: 'b05c6c5221b307ad41484ee527fa127ad1a09a0818ec64775309a3d4e4d40143',
+    stacks_tip_burn_block: undefined,
+    exit_at_block_height: null
+  },
+  blocktimeInfo: {
+    mainnet: { target_block_time: 600 },
+    testnet: { target_block_time: 120 }
+  }
+}
+```
+
+-> Stacking execution will differ between mainnet and testnet in terms of cycle times and participation thresholds
+
+With the obtained PoX info, you can present whether Stacking has been executed in the next cycle, when the next cycle begins, the duration of a cycle, and the minimum microstacks required to participate:
+
+```js
+// will Stacking be executed in the next cycle?
+const stackingExecution = poxInfo.rejection_votes_left_required > 0;
```

Comment:
`rejection_votes_left_required` isn't returned currently

Reply:
checking with @lgalabru. I was under the impression it would be returned

Reply:
Correct, will be addressed with https://github.com/blockstack/stacks-blockchain/pull/1920

Reply:
PR merged, feature will be available after next deployment.